mailmerge/template_message.py

```diff
@@ -120,38 +128,52 @@ def _transform_markdown(self):
 
     def _transform_attachments(self):
```

Comment:
It would be great to get your feedback on this refactor @seshrs !

Reply:
Changes include:
- Move file path resolution to a helper function
- Use pathlib `Path` objects instead of strings
- Add comments